x-pack/plugins/task_manager/server/task_store.ts

```diff
@@ -438,14 +473,20 @@ export class TaskStore {
   private async search(opts: SearchOpts = {}): Promise<FetchResult> {
     const { query } = ensureQueryOnlyReturnsTaskObjects(opts);
 
-    const result = await this.callCluster('search', {
-      index: this.index,
-      ignoreUnavailable: true,
-      body: {
-        ...opts,
-        query,
-      },
-    });
+    let result;
+    try {
+      result = await this.callCluster('search', {
```

Comment:
>I wonder if we can pick out 429s out of there reasonably well 

I enhanced the saved objects errors helper to hopefully facilitate this in #75664. To detect SOC 429 errors, it would be done via `SavedObjectsErrorHelpers.isTooManyRequestsError(e)`. I'm not too sure yet how it will look on callCluster calls, I've noticed some include retry logic instead of throwing 429 errors so I'll have to see :-) 

I think the same as you said, we'll see how it looks like at the next step and we can improve if needed.